### PR TITLE
VIH-5405 implement base class update status

### PR DESCRIPTION
**Pull request by valentinabennett**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/VIH-5405

### Change description ###

implemented base class to update status to Not signed in

**Does this PR introduce a breaking change?** (check one with "x")

```
[ X] Yes
[ ] No
```

### CODE REVIEW

File: VideoWeb/VideoWeb/ClientApp/src/app/on-the-day/participant-self-test/participant-self-test.component.ts [hunk 1]
```diff
@@ -38,4 +41,22 @@ export class ParticipantSelfTestComponent extends BaseSelfTestComponent {
         this.logger.debug('restarting participant self-test');
         this.selfTestComponent.replayVideo();
     }
+
+    @HostListener('window:beforeunload', ['$event'])
```
Comment (shaed-parkar) [inline VideoWeb/VideoWeb/ClientApp/src/app/on-the-day/participant-self-test/participant-self-test.component.ts:45]: Could you add tests for this new set of code, including negative scenarios

File: VideoWeb/VideoWeb/ClientApp/src/app/on-the-day/models/participant-status-base.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+        try {
+            const eventConferenceId = this.route.snapshot.paramMap.get('conferenceId');
+
+            this.participantStatusUpdateService
+                .postParticipantStatus(EventType.ParticipantNotSignedIn, eventConferenceId)
+                .then(() => {
+                    this.logger.info('Participant status was updated to not signed in');
+                })
+                .catch(err => {
+                    this.logger.error('Unable to update status to not signed in', err);
```
Comment (shaed-parkar) [inline VideoWeb/VideoWeb/ClientApp/src/app/on-the-day/models/participant-status-base.ts:24]: Can you add tests for negative scenarios

File: VideoWeb/VideoWeb/ClientApp/src/app/on-the-day/models/participant-status-base.spec.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+describe('ParticipantStatusBase', () => {
+    const activatedRoute: any = { snapshot: { paramMap: convertToParamMap({ conferenceId: '1234567' }) } };
+    const loggerMock: Logger = new MockLogger();
+    let participantStatusUpdateServiceSpy: jasmine.SpyObj<ParticipantStatusUpdateService>;
+    participantStatusUpdateServiceSpy = jasmine.createSpyObj('ParticipantStatusUpdateService', ['postParticipantStatus']);
+
+    const component = new ParticipantStatusBaseTest(participantStatusUpdateServiceSpy, loggerMock, activatedRoute);
+
+    it('should update participant status on log out', () => {
+        const event: any = { returnValue: 'save' };
```
Comment (shaed-parkar) [inline VideoWeb/VideoWeb/ClientApp/src/app/on-the-day/models/participant-status-base.spec.ts:27]: this code doesn't wait for the code to complete so the assertion for the positive result is missing (i.e. verify log was called with the right message)

Comment (valentinabennett) [reply VideoWeb/VideoWeb/ClientApp/src/app/on-the-day/models/participant-status-base.spec.ts:27]: done

File: VideoWeb/VideoWeb/ClientApp/src/app/on-the-day/models/participant-status-base.ts (hunk 2 above)
Comment (valentinabennett) [reply VideoWeb/VideoWeb/ClientApp/src/app/on-the-day/models/participant-status-base.ts:24]: done

File: VideoWeb/VideoWeb/ClientApp/src/app/on-the-day/participant-self-test/participant-self-test.component.ts (hunk 1 above)
Comment (valentinabennett) [reply VideoWeb/VideoWeb/ClientApp/src/app/on-the-day/participant-self-test/participant-self-test.component.ts:45]: done

File: VideoWeb/VideoWeb/ClientApp/src/app/on-the-day/models/participant-status-base.ts [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+export abstract class ParticipantStatusBase {
+    constructor(
+        protected participantStatusUpdateService: ParticipantStatusUpdateService,
+        protected logger: Logger,
+        protected route: ActivatedRoute
+    ) {}
+
+    raiseNotSignedIn() {
+        try {
```
Comment (shaed-parkar) [inline VideoWeb/VideoWeb/ClientApp/src/app/on-the-day/models/participant-status-base.ts:15]: this try-catch block won't work since the promise is already handling errors on line 23.

Comment (valentinabennett) [reply VideoWeb/VideoWeb/ClientApp/src/app/on-the-day/models/participant-status-base.ts:15]: there is not only promise in this block of code.

File: VideoWeb/VideoWeb/ClientApp/src/app/on-the-day/models/participant-status-base.ts [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+            const eventConferenceId = this.route.snapshot.paramMap.get('conferenceId');
```
Comment (shaed-parkar) [inline VideoWeb/VideoWeb/ClientApp/src/app/on-the-day/models/participant-status-base.ts:16]: Each of the participant classes retrieve the conferenceId upon entry.
There is no need to handle this on exit. It's a safe assumption to expect this line to pass.

I suggest an abstract variable for each of the child classes to inherit is the safest option

Comment (valentinabennett) [reply VideoWeb/VideoWeb/ClientApp/src/app/on-the-day/models/participant-status-base.ts]: Done
